Compression/src/CompressDetStepMCs_module.cc

```diff
@@ -259,29 +261,31 @@ void mu2e::CompressDetStepMCs::produce(art::Event & event)
     // then compressStepPointMCs now
     compressStepPointMCs(event);
   }
-  if (_mcTrajectoryCompressionLevel == mu2e::CompressionLevel::kSimParticleCompression) {
-    // if we are compressing MCTrajectories based on the SimParticles we are keeping,
-    // then compress MCTrajectories now
-    compressMCTrajectories(event);
+  if (_conf.mcTrajectoryTag() != "") {
+    if (_mcTrajectoryCompressionLevel == mu2e::CompressionLevel::kSimParticleCompression) {
+      // if we are compressing MCTrajectories based on the SimParticles we are keeping,
+      // then compress MCTrajectories now
+      compressMCTrajectories(event);
+    }
   }
 
   // Update all the data products so that their SimParticlePtrs point to the new collection
-  updateStrawGasSteps();
-  updateCaloShowerSteps();
-  updateCrvSteps();
+  if (_conf.strawGasStepTag() != "") { updateStrawGasSteps(); }
+  if (_conf.caloShowerStepTag() != "") { updateCaloShowerSteps(); }
+  if (_conf.crvStepTag() != "") { updateCrvSteps(); }
   updateStepPointMCs();
-  updateMCTrajectories();
+  if (_conf.mcTrajectoryTag() != "") { updateMCTrajectories(); }
 
   // Now add everything to the event
-  event.put(std::move(_newStrawGasSteps));
-  event.put(std::move(_newCaloShowerSteps));
-  event.put(std::move(_newCrvSteps));
+  if (_conf.strawGasStepTag() != "") { event.put(std::move(_newStrawGasSteps)); }
+  if (_conf.caloShowerStepTag() != "") { event.put(std::move(_newCaloShowerSteps)); }
+  if (_conf.crvStepTag() != "") { event.put(std::move(_newCrvSteps)); }
   for (const auto& i_tag : _conf.stepPointMCTags()) {
     event.put(std::move(_newStepPointMCs.at(i_tag.instance())), i_tag.instance());
   }
   event.put(std::move(_newSimParticles));
   event.put(std::move(_newGenParticles));
-  event.put(std::move(_newMCTrajectories));
+  if (_conf.mcTrajectoryTag() != "") { event.put(std::move(_newMCTrajectories)); }
```

Comment:
Making the output collections predicate on the input being present will break downstream workflows that depend on a consistent set of products.  I know this change was made in response to a reviewer request, but I ask that you back out this logic, so that the outputs are always created, even if there are no corresponding inputs.  The logic on the inputs can stay.

Reply:
That makes sense. I've made these changes in commit e8e0ca6

Reply:
> Making the output collections predicate on the input being present will break downstream workflows that depend on a consistent set of products.

Of course it won't, because in the workflow you have in mind the input
tags will never be empty.  And for a workflow where a product is
missing, it makes zero sense to create it as a "compression" step.
The "ideal" modular solution would not create stuff out of nothing.

I am giving up here and will simply accept this PR, no more requests.